test/integration/suite.go

```diff
@@ -27,6 +27,12 @@ import (
 	"github.com/kong/gateway-operator/test/helpers/certificate"
 )
 
+func init() {
+	var err error
+	sharedHTTPClient, err = helpers.CreateHTTPClient(nil, "")
+	exitOnErr(err)
+}
```

Comment:
Is there a specific reason for this?

I'd generally be a bit hesitant to share objects between the tests unless absolutely necessary. It's fine to use the shared client here but what's the goal for this?

I'd say we should strive to make tests more isolated rather than keep adding shared resources to them.